### ISSUE TITLE
Bug: `vim.lsp.diagnostic.get_count()` is deprecated

### ISSUE THREAD

**Issue by numine777** [nvim-lualine/lualine.nvim#473]
### Self Checks

- [X] I'm using the latest lualine.
- [X] I didn't find the issue in exsisting issues or prs.

### How to reproduce the problem
1. Update to neovim nightly
2. Open neovim with lualine plugin

### Expected behaviour
Neovim functions normally with lualine integration

### Actual behaviour
A persistent notification for `vim.lsp.diagnostic.get_count() is deprecated` will appear.

### Minimal config to reproduce the issue
`require 'lualine'.setup()`

<details>
<summary>Config</summary>

</details>

### Aditional information
This is root caused to `lua/lualine/components/diagnostics/sources.lua`.  I will work on setting up a PR to solve this issue, but wanted to bring awareness.

**Comment by numine777** [comment]
Pull request https://github.com/nvim-lualine/lualine.nvim/pull/474 solves the issue

**Comment by shadmansaleh** [comment]
use the `nvim` source instead of `nvim-lsp` for diagnostics.

**Comment by numine777** [comment]
> Use the `nvim` source instead of `nvim-lsp` for diagnostics.

That's what my PR does 😉

**Comment by shadmansaleh** [comment]
> That's what my PR does 😉

No your pr is just implementing the functionality that already exists in `nvim` diagnostics source to `nvim-lsp` diagnostics source .

**Comment by numine777** [comment]
> > That's what my PR does 😉
> 
> No your pr is just implementing the functionality that already exists in [nvim](https://github.com/nvim-lualine/lualine.nvim/blob/d68631d2c02bd31d937349d739c625cc81dd9ac1/lua/lualine/components/diagnostics/sources.lua#L14-L24) diagnostics source to `nvim-lsp` diagnostics source 😄

Got it, thanks!

**Comment by flaviusbuffon** [comment]
When will you solve this problem?
Thanks！

**Comment by shadmansaleh** [comment]
> When will you solve this problem?

TBH solution is just to change default config .

For now any one facing this issue just need to slightly change there config like following.

```diff
- { 'diagnostics', sources = {'nvim-lsp'}}
+ { 'diagnostics', sources = {'nvim'}}
```

**Comment by GaetanLepage** [comment]
No, I still have the deprecation warning...

**Comment by gegoune** [comment]
There is bunch of plugins that did not get updated to avoid deprecation warning yet. lualine is not one of them (given `nvim` source is used).

**Comment by shadmansaleh** [comment]
> There is bunch of plugins that did not get updated to avoid deprecation warning yet.

The deprecation commit was just made yesterday https://github.com/neovim/neovim/commit/c4d70dae802ef074aaf54bdcbbd5f73380f74a86 in nightly stable releases are perfectly fine. You guys can't be following nightly that closely and not expect breakages. I don't think any plugin commit that they will be supporting nightly . TBH no one commited to supporting latest stable release either.

For example here's what MIT license has to say about it
https://github.com/nvim-lualine/lualine.nvim/blob/d68631d2c02bd31d937349d739c625cc81dd9ac1/LICENSE#L15
```
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
```

There's no commitment for any updates what so ever . All popular OSS licenses have similar clause in them.

Basically what I want to say is if you want a stable experience use stable not nightly. If you're using nightly expect breakages (you've signed up for it by installing nightly) . When breakages happen give plugin authors some time to fix them they are working to on these mostly as hobby projects.

Note: This is not directed toward any individual but rather any one who is expecting nightly to be stable .

**Comment by shadmansaleh** [comment]
You can close it once it's fixed 😄

**Comment by numine777** [comment]
Sorry about that, didn't realize it was still an issue.  I can help out if there's a way to reproduce the issue!  Mine stopped complaining with the change so I assumed that the issue was resolved.
